### PR TITLE
Improve /api/v1/repos/issues/search by just getting repo ids

### PR DESCRIPTION
**Pull request by zeripath**
/api/v1/repos/issues/search is a highly inefficient search which is unfortunately
the basis for our dependency searching algorithm. In particular it currently loads
all of the repositories and their owners and their primary coding language all of
which is immediately thrown away.

This PR makes one simple change - just get the IDs.

Related #14560
Related #12827

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Comment by 6543** [comment]
.

**Review by 6543 — APPROVED** [review]

File: models/repo_list.go [hunk 1]
```diff
@@ -376,31 +406,23 @@ func SearchRepositoryByCondition(opts *SearchRepoOptions, cond builder.Cond, loa
 	}
 
 	sess := x.NewSession()
-	defer sess.Close()
```
Comment (lunny) [inline models/repo_list.go:379]: This line should not be removed.

File: models/repo_list.go [hunk 2]
```diff
@@ -376,31 +406,23 @@ func SearchRepositoryByCondition(opts *SearchRepoOptions, cond builder.Cond, loa
...
-	if err != nil {
-		return nil, 0, fmt.Errorf("Count: %v", err)
+	var count int64
+	if opts.PageSize > 0 {
+		var err error
+		count, err = sess.
+			Where(cond).
+			Count(new(Repository))
+		if err != nil {
+			return nil, 0, fmt.Errorf("Count: %v", err)
```
Comment (lunny) [inline models/repo_list.go:418]: then sess is not Closed

Comment (zeripath) [reply models/repo_list.go:418]:
```suggestion
			_ = sess.Close()
			return nil, 0, fmt.Errorf("Count: %v", err)
```

**Review by noerw — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Pull request merged**